pkg/sidecars/server/server.go

```diff
@@ -0,0 +1,115 @@
+package sidecarserv
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/inspr/inspr/pkg/sidecars/models"
+)
+
+// Server is a struct that contains the variables necessary
+// to handle the necessary routes of the rest API
+type Server struct {
+	broker       string
+	Reader       models.Reader
+	Writer       models.Writer
+	inAddr       string
+	outAddr      string
+	client       *http.Client
+	runningRead  bool
+	runningWrite bool
+}
+
+// NewServer returns a new sidecar server
+func NewServer() *Server {
```

Comment:
You can remove this method, and just instantiate a new server within `Init(...)`